inference-engine/thirdparty/clDNN/src/graph_optimizer/prepare_primitive_fusing.cpp

```diff
@@ -490,13 +490,28 @@ void prepare_primitive_fusing::fuse_simple_primitives(program_impl &p) {
 
             std::vector<bool> can_fuse_parents = { can_fuse_parent1, can_fuse_parent2 };
 
+            auto p1_raw_size = parent1->get_output_layout().size.raw;
+            auto p2_raw_size = parent2->get_output_layout().size.raw;
+            for (unsigned k = 0; k < p1_raw_size.size(); k++) {
+                if (p1_raw_size[k] < p2_raw_size[k]) {
+                    if (p1_raw_size[k] != 1)
+                        return;
+                    can_fuse_parents[0] = false;
+                }
+                if (p2_raw_size[k] < p1_raw_size[k]) {
```

Comment:
Maybe "else if (p2_raw_size[k] < p1_raw_size[k]) { ... " ? 
Because you don't have to check this condition for all cases.

Reply:
Done